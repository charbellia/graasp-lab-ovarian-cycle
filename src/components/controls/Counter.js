import React from 'react';

export const Counter = ({ handleStart, handleStop }) => (
  <div className="Counter-Container">
    <div className="clock-container">
      <div className="clock-balls">
        <div className="clock-timer">
          <span className="clock-counter"> 01 </span>
          <span className="clock-dayer"> JOURS </span>
        </div>
      </div>
      <div className="clock-setter">
        <button className='button-start' onClick={handleStart}>START</button>
        <button className='button-stop' onClick={handleStop}>STOP</button>
      </div>
    </div>
  </div>
);

export default (Counter);
