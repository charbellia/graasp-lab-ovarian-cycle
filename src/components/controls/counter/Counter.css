.clock-starter {
  text-align: center;
  padding: 10px;
}
.fa-calendar {
  width: 15rem;
  margin-bottom: 2rem;
  margin-top: 2rem;
}
.btn-lg {
  font-size: 2rem;
}
