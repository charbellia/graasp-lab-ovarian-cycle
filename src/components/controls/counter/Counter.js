import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './Counter.css';
import { Button } from 'reactstrap';
import { withNamespaces } from 'react-i18next';

// deciding what happen on the launch/stop cycle button cycle
// the t is used for the translation
export const Counter = ({
  handleFullCycle,
  handleStop,
  isStarted,
  ovulationActive,
  preOvulationActive,
  postOvulationActive,
  t,
  themeColor,
}) => {
  const isDisabled = preOvulationActive || ovulationActive || postOvulationActive;
  // show stop button if simulation started otherwise show start
  const startButton = (isStarted)
    ? (
      <Button
        onClick={handleStop}
        size="lg"
        disabled={isDisabled}
        style={{ backgroundColor: themeColor, borderColor: themeColor }}
        className="btn-full-cycle"
      >
        {t('Stop Cycle')}
      </Button>
    )
    : (
      <Button
        onClick={handleFullCycle}
        size="lg"
        disabled={isDisabled}
        style={{ backgroundColor: themeColor, borderColor: themeColor }}
        className="btn-full-cycle"
      >
        {t('Full Cycle')}
      </Button>
    );
  return (
    <div className="counter-container">
      <div className="description-content">
        <h2>{t('Description')}</h2>
        <p>
          {t('Lab Explanation')}
        </p>
      </div>
      <div className="clock-setter">
        {startButton}
        <hr className="separator-line" />
      </div>
    </div>
  );
};

Counter.propTypes = {
  handleFullCycle: PropTypes.func.isRequired,
  handleStop: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  isStarted: PropTypes.bool.isRequired,
  ovulationActive: PropTypes.bool.isRequired,
  postOvulationActive: PropTypes.bool.isRequired,
  preOvulationActive: PropTypes.bool.isRequired,
  themeColor: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  themeColor: state.layout.themeColor,
  preOvulationActive: state.simulation.preOvulationActive,
  postOvulationActive: state.simulation.postOvulationActive,
  ovulationActive: state.simulation.ovulationActive,
});

const ConnectedComponent = connect(mapStateToProps)(Counter);

export default withNamespaces()(ConnectedComponent);
