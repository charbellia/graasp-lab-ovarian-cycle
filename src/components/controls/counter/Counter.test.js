import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import Counter from './Counter';

describe('<Counter />', () => {
  const props = {
    t: jest.fn(),
    handleFullCycle: jest.fn(),
    handleStop: jest.fn(),
    isStarted: false,
    ovulationActive: false,
    postOvulationActive: false,
    preOvulationActive: false,
    isDisabled: false,
  };

  const component = shallow(<Counter {...props} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('has one <div /> with class counter-container', () => {
    expect(component.find('div.counter-container').length).toEqual(1);
  });
  it('shows one <Button />', () => {
    expect(component.find(Button).length).toEqual(1);
  });
});
