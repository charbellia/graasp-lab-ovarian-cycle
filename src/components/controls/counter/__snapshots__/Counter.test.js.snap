// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    handleFullCycle={[MockFunction]}
    handleStop={[MockFunction]}
    isStarted={false}
    ovulationActive={false}
    postOvulationActive={false}
    preOvulationActive={false}
    t={
      [MockFunction] {
        "calls": Array [
          Array [
            "Full Cycle",
          ],
          Array [
            "Description",
          ],
          Array [
            "Lab Explanation",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    themeColor="#5050d2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="description-content"
        >
          <h2 />
          <p />
        </div>,
        <div
          className="clock-setter"
        >
          <Button
            className="btn-full-cycle"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="lg"
            style={
              Object {
                "backgroundColor": "#5050d2",
                "borderColor": "#5050d2",
              }
            }
            tag="button"
          />
          <hr
            className="separator-line"
          />
        </div>,
      ],
      "className": "counter-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <p />,
          ],
          "className": "description-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="btn-full-cycle"
              color="secondary"
              disabled={false}
              onClick={[MockFunction]}
              size="lg"
              style={
                Object {
                  "backgroundColor": "#5050d2",
                  "borderColor": "#5050d2",
                }
              }
              tag="button"
            />,
            <hr
              className="separator-line"
            />,
          ],
          "className": "clock-setter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "btn-full-cycle",
              "color": "secondary",
              "disabled": false,
              "onClick": [MockFunction],
              "size": "lg",
              "style": Object {
                "backgroundColor": "#5050d2",
                "borderColor": "#5050d2",
              },
              "tag": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "separator-line",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="description-content"
          >
            <h2 />
            <p />
          </div>,
          <div
            className="clock-setter"
          >
            <Button
              className="btn-full-cycle"
              color="secondary"
              disabled={false}
              onClick={[MockFunction]}
              size="lg"
              style={
                Object {
                  "backgroundColor": "#5050d2",
                  "borderColor": "#5050d2",
                }
              }
              tag="button"
            />
            <hr
              className="separator-line"
            />
          </div>,
        ],
        "className": "counter-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <p />,
            ],
            "className": "description-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="btn-full-cycle"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                size="lg"
                style={
                  Object {
                    "backgroundColor": "#5050d2",
                    "borderColor": "#5050d2",
                  }
                }
                tag="button"
              />,
              <hr
                className="separator-line"
              />,
            ],
            "className": "clock-setter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "className": "btn-full-cycle",
                "color": "secondary",
                "disabled": false,
                "onClick": [MockFunction],
                "size": "lg",
                "style": Object {
                  "backgroundColor": "#5050d2",
                  "borderColor": "#5050d2",
                },
                "tag": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "separator-line",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
